core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -417,6 +417,13 @@ class ZkUtils(val zkClient: ZkClient,
     topicDirs.consumerOwnerDir + "/" + partition
   }
 
+  def controllerZkData(brokerId: Int, timestamp: Long): String = {
+    Json.encode(Map("version" -> 1, "brokerid" -> brokerId, "timestamp" -> timestamp.toString))
+  }
+
+  def preferredReplicaLeaderElectionZkData(partitions: scala.collection.Set[TopicAndPartition]): String = {
```

Comment:
Updated the PR moving the two new methods into the companion object.

I think ideally all znode data definitions would've been in the ZkUtils companion object to begin with instead of being spread across random classes and the ZkUtils class. Something I just noticed: apparently while you can call static methods from java class instances, you cannot call companion object methods from scala class instances. Because of this, I'm hesitant to move the definitions from the ZkUtils class over to the ZkUtils companion object as it could break a lot of external code that uses ZkUtils.